{
   "active_lock_reason" : null,
   "assignee" : null,
   "assignees" : [],
   "author_association" : "CONTRIBUTOR",
   "body" : "The test fails intermittently, see https://cirrus-ci.com/task/6403578080788480?logs=ci#L3521 and https://github.com/bitcoin/bitcoin/pull/24748#issuecomment-1916996716.\r\nI think it's because of a race between the python NetworkThread and the actual\r\ntest, which will both call `initiate_v2_handshake`. I could reproduce it by adding a sleep into `initiate_v2_handshake` after the line `self.sent_garbage = random.randbytes(garbage_len)`.\r\n\r\nFix this by waiting for the first `initiate_v2_handshake` to have finished before calling it a second time.",
   "closed_at" : null,
   "closed_by" : null,
   "comments" : 1,
   "comments_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/29352/comments",
   "created_at" : "2024-01-30T23:06:03Z",
   "draft" : false,
   "events_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/29352/events",
   "html_url" : "https://github.com/bitcoin/bitcoin/pull/29352",
   "id" : 2108959832,
   "labels" : [
      {
         "color" : "d4c5f9",
         "default" : false,
         "description" : null,
         "id" : 62963516,
         "name" : "Tests",
         "node_id" : "MDU6TGFiZWw2Mjk2MzUxNg==",
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/labels/Tests"
      }
   ],
   "labels_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/29352/labels{/name}",
   "locked" : false,
   "milestone" : null,
   "node_id" : "PR_kwDOABII585lgrz9",
   "number" : 29352,
   "performed_via_github_app" : null,
   "pull_request" : {
      "diff_url" : "https://github.com/bitcoin/bitcoin/pull/29352.diff",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/29352",
      "merged_at" : null,
      "patch_url" : "https://github.com/bitcoin/bitcoin/pull/29352.patch",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/29352"
   },
   "reactions" : {
      "+1" : 0,
      "-1" : 0,
      "confused" : 0,
      "eyes" : 0,
      "heart" : 0,
      "hooray" : 0,
      "laugh" : 0,
      "rocket" : 0,
      "total_count" : 0,
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/29352/reactions"
   },
   "repository_url" : "https://api.github.com/repos/bitcoin/bitcoin",
   "state" : "open",
   "state_reason" : null,
   "timeline_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/29352/timeline",
   "title" : "test: fix intermittent failure in p2p_v2_earlykeyresponse",
   "updated_at" : "2024-01-30T23:06:07Z",
   "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/29352",
   "user" : {
      "avatar_url" : "https://avatars.githubusercontent.com/u/48763452?v=4",
      "events_url" : "https://api.github.com/users/mzumsande/events{/privacy}",
      "followers_url" : "https://api.github.com/users/mzumsande/followers",
      "following_url" : "https://api.github.com/users/mzumsande/following{/other_user}",
      "gists_url" : "https://api.github.com/users/mzumsande/gists{/gist_id}",
      "gravatar_id" : "",
      "html_url" : "https://github.com/mzumsande",
      "id" : 48763452,
      "login" : "mzumsande",
      "node_id" : "MDQ6VXNlcjQ4NzYzNDUy",
      "organizations_url" : "https://api.github.com/users/mzumsande/orgs",
      "received_events_url" : "https://api.github.com/users/mzumsande/received_events",
      "repos_url" : "https://api.github.com/users/mzumsande/repos",
      "site_admin" : false,
      "starred_url" : "https://api.github.com/users/mzumsande/starred{/owner}{/repo}",
      "subscriptions_url" : "https://api.github.com/users/mzumsande/subscriptions",
      "type" : "User",
      "url" : "https://api.github.com/users/mzumsande"
   }
}
